src/api/BaseAPI.ts

```diff
@@ -0,0 +1,38 @@
+import HTTPTransport from 'util/HTTPTransport';
+
+abstract class BaseAPI {
+  protected http: HTTPTransport;
+
+  protected constructor(endpoint: string) {
+    this.http = new HTTPTransport(endpoint);
+  }
+}
+
+interface CreateApiMethod {
+  create(data: unknown): Promise<unknown>
+}
+
+interface StoreApiMethod {
+  store(data: unknown): Promise<unknown>
+}
+
+interface ReadApiMethod {
+  read(data: unknown): Promise<unknown>
```

Comment:
этот метод можно типизировать с помощью generic'ов, например так:
```suggestion
  read<TResponse>(data: unknown): Promise<TResponse>
```
тогда, если мы знает тип ответа от сервера, мы сможем более конкретно типизировать запрос